### ISSUE TITLE
HttpClientFactory waits for free connection, causing latency spikes.

### ISSUE THREAD

**Issue by LosGlennos** [aspnet/Extensions#1755]
HttpClientFactory waits for a connection for 1 second before firing request.

Using version '2.2' of package 'Microsoft.AspNetCore.Mvc'

Have not able to reproduce error locally, We have added `services.AddHttpClient()` with no additional configuration in the `Startup.cs`. We are serving some 1 million requests a day.

Stack trace and execution time from a profiling made in Application Insights:

![image](https://user-images.githubusercontent.com/2778229/58425097-43f51e80-8099-11e9-9245-d11a775164d1.png)

(Sorry for the task bar in the middle of everything)

This seems to be happening around every 2 minutes.

We have tried bumping the version from 2.1 to 2.2 with no luck.

**Comment by rynowak** [comment]
You're seeing the time taken to establish a connection each time the pooled handler is recycled. The default timeout of a handler is 2 mins. You can configure the timeout using  https://github.com/aspnet/Extensions/blob/master/src/HttpClientFactory/Http/src/HttpClientFactoryOptions.cs#L58

Let me know if you have something else in mind. 

If you're interested in understanding why creating new connections is taking 1 second, then this will be possible to reproduce without using the factory.

**Comment by LosGlennos** [comment]
Hi @rynowak 

Thank you for your response. We bumped the recycle time to 60 minutes, however the issue persisted. But we did managed to solve this yesterday!

It's related to this issue https://github.com/dotnet/corefx/issues/24104

By doing DNS lookup to localhost it sometimes fails to resolve the IPv6-adress and then it apparently takes ~1 second to connect and fire a request to IPv4. So we do all our internal communication now towards 127.0.0.1, that way we don't strike the failed ::1 IPv6-adress.

I'll close this issue as it is not a HttpClientFactory-problem.
